joomla/joomla-cms
# [4.0] Changes to bootstrap-init.js to support joomla:updated event

Pull Request for Issue # .

### Summary of Changes

Refactored the code into a function, and added events handlers for DOM ready and joomla:updated.  In order to allow the code to be re-entrant, need to remove the options after BS init has been run, so we don't add multiple event delegations.

Also tidied up a few missing arguments, unused variables, and changed some 'var' to 'let'.

### Testing Instructions

Basic testing is to load a page with Bootstrap widgets on it, like anything with tabs on the backend, and observe that everything still works.  Testing the joomla:updated event isn't really possible, as it's designed to allow adding new forms to the DOM on the fly.  You can prove that the code runs by putting a breakpoint somewhere in the _initBootstrap function, then fire the event in the console with ...

Joomla.Event.dispatch(document, 'joomla:updated')

... and you'll hit your breakpoint.  But without adding a new form to the DOM and adding the options for it, which is hard to provide a test case for, that won't actually prove it works.  But it does.  My extension (Fabrik) uses this extensively, and I can confirm it does work.

### Expected result



### Actual result



### Documentation Changes Required
